src/Analyzer/DocBlockNameResolver.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Arkitect\Analyzer;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Use_;
+use PhpParser\NodeVisitor\NameResolver;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+
+class DocBlockNameResolver extends NameResolver
+{
+    public function enterNode(Node $node): void
+    {
+        if ($node instanceof Use_) {
+            foreach ($node->uses as $use) {
+                $this->addAlias($use, $node->type, null);
+            }
+        } elseif ($node instanceof Node\Stmt\Property) {
+            $lexer = new Lexer();
+            $typeParser = new TypeParser();
+            $constExprParser = new ConstExprParser();
+            $phpDocParser = new PhpDocParser($typeParser, $constExprParser);
+
+            if (null === $node->getDocComment()) {
+                return;
+            }
+
+            /** @var Doc $docComment */
+            $docComment = $node->getDocComment();
+
+            $tokens = $lexer->tokenize($docComment->getText());
+            $tokenIterator = new TokenIterator($tokens);
+            $phpDocNode = $phpDocParser->parse($tokenIterator);
+
+            foreach ($phpDocNode->getVarTagValues() as $tagValue) {
+                $type = $this->resolveName(new Node\Name((string) $tagValue->type), Use_::TYPE_NORMAL);
+                $node->type = $type;
+                break;
```

Comment:
This break I don't know if this is correct or not, it seems that the first element is the type that we want.  I haven't found cases with more elements, but to avoid problems  I preferred to exit immediately.